presto-hive/src/main/java/com/facebook/presto/hive/HiveWriterFactory.java

```diff
@@ -332,6 +332,12 @@ public HiveWriter createWriter(Page partitionColumns, int position, OptionalInt
                     updateMode = UpdateMode.NEW;
                     writeInfo = locationService.getPartitionWriteInfo(locationHandle, partition, partitionName.get());
                 }
+                else if (table.isTemporary()) {
+                    // this assumes that temporary table is always empty at this step
```

Comment:
hmm.. didn't really understand the comment :) . Looks like even if temporary table is not empty, use `APPEND` update mode also makes sense.

Or shall we say `Note: temporary table is always empty at this step...` ?